R/options.R

```diff
@@ -1,4 +1,41 @@
-default_tty_tick_interval <- function() {
-  # refresh interval in milliseconds
-  getOption(paste0(utils::packageName(), ".tty_tick_interval"), 100) / 1000
-}
+#' @import options
+options::define_options(
+  "tty refresh interval when reporting results in milliseconds",
+  tty_tick_interval = 0.1,
+  
+  "character vector indicating whether R error should be thrown when issues
+   are discovered when generating results. \"never\" means that no errors
+   are thrown. If \"issues\" then errors are emitted only on issues, whereas
+   \"potential issues\" stands for error on both issues and potential issues.",
+  results_error_on = "never",
+  
+  "character vector indicating which packages should be included in the results.
+   \"all\" means that all packages are kept. If \"issues\" then only packages 
+   with issues identified, whereas \"potential_issues\" stands for keeping
+   packages with both \"issues\" and \"potential_issues\".",
+  results_keep = "all",
+  
+  "`logical` indicating whether output directory should be unlinked before
+   running checks. If `FALSE`, an attempt will me made to restore previous
+   progress from the same `output`",
+  restore = NA,
+  
+  "`logical` indicating whether default R CMD check related enviromental
+   variables should be appended.",
+  default_check_env_variables = TRUE,
+
+  "`logical` indicating whether default R CMD check build args should be appended.",
+  default_check_build_args = TRUE,
+
+  "`logical` indicating whether default R CMD check args should be appended.",
+  default_check_args = TRUE
```

Comment:
Instead of having a logical flag, then embedding some defaults in the package, we should just expose the defaults.

```suggestion
  "R CMD check related environment variables",
  default_check_env_variables = c(
    "_R_CHECK_FORCE_SUGGESTS_" = FALSE,
    "_R_CHECK_RD_XREFS_" = FALSE,
    "_R_CHECK_SYSTEM_CLOCK_" = FALSE,
    "_R_CHECK_SUGGESTS_ONLY_" = TRUE
  ),

  "R CMD check build args",
  default_check_build_args = c(
    "--no-build-vignettes",
    "--no-manual"
  ),

  "R CMD check args",
  default_check_args =c(
    "--timings",
    "--ignore-vignettes",
    "--no-manual"
  )
```

Reply:
Honestly, I'd like to avoid that. These default options ensure that the checks are sane by default (for instance,`"_R_CHECK_SUGGESTS_ONLY_", which forces dependencies isolation). If we make them exposed defaults, then whenever someone adds their own option, it would overwrite defaults, which almost always should be used. Therefore, I'd like to stick to the mechanism when the user appends to the existing options, except when explicitly set otherwise.

Reply:
I think this is a matter of what we feel should be the purpose of `checked`. Personally, I want `checked` to be the best `R CMD check` orchestration tool. We have opinions on how to run `R CMD check` most rigorously, but I don't think it's the role of the tool to constrain how people choose to run checks themselves. 

This is why I want opinionated defaults, but to be flexible enough for other use cases.

If we assume that anyone who is investigating global flags to configure the tool is a savvy user, then I think we can trust them with the freedom to use the tool as they see fit.

If we're concerned about people overwriting defaults with flags that produce behaviors that they're not intending, then I think the solution is effective documentation to make the intention clear, not restricting the use of those flags.

Reply:
Fair. Considering it and the online discussion we had, I've implemented options according to your suggestions.